crates/sui-types/src/signature.rs

```diff
@@ -73,10 +77,13 @@ impl ToFromBytes for GenericSignature {
                 | SignatureScheme::Secp256r1 => Ok(GenericSignature::Signature(
                     Signature::from_bytes(bytes).map_err(|_| FastCryptoError::InvalidSignature)?,
                 )),
-                SignatureScheme::MultiSig => {
-                    let multisig = MultiSig::from_bytes(bytes)?;
-                    Ok(GenericSignature::MultiSig(multisig))
-                }
+                SignatureScheme::MultiSig => match MultiSig::from_bytes(bytes) {
+                    Ok(multisig) => Ok(GenericSignature::MultiSig(multisig)),
```

Comment:
for the rare (or impossible?) case that `bytes` is equivocal - maybe the default should be MultiSigLegacy and fallback to MultiSig? it's safer in case some validators have not upgraded (and then they only process this as MultiSigLegacy).

Reply:
but should we try the latest deser first is to make it faster since this is what we want to provide first class support for? also this is what we plan to use in the long term